src/Fixer/Whitespace/BlankLineBeforeControlStatementFixer.php

```diff
@@ -0,0 +1,240 @@
+                continue;
+            }
+
+            $prevToken = $tokens[$index - 1];
+
+            if ($prevToken->isWhitespace()) {
+                $parts = explode("\n", $prevToken->getContent());
+                $countParts = count($parts);
+
+                if (1 === $countParts) {
+                    $prevToken->setContent(rtrim($prevToken->getContent(), " \t").$lineEnding.$lineEnding);
+                } elseif (count($parts) <= 2) {
+                    $prevToken->setContent($lineEnding.$prevToken->getContent());
+                }
+            } else {
+                $tokens->insertAt($index, new Token(array(T_WHITESPACE, $lineEnding.$lineEnding)));
+
+                ++$index;
+                ++$limit;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'An empty line feed should precede a control statement.',
+            array(
+                new CodeSample(
+'<?php 
+                    
+function A() {
+    echo 1;
+    return 1;
+}'
+                ),
+                new CodeSample(
+                '<?php 
+                    
+switch ($foo) { 
+    case 42:
+        $bar->process();
+        break;
+    case 44:
+        break;
+}',
+                    array(
+                        'break',
+                    )
+                ),
+                new CodeSample(
+'<?php 
+                    
+foreach ($foo as $bar) {
+    if ($bar->isTired()) {
+        $bar->sleep();
+        continue;
+    }
+}',
+                    array(
+                        'continue',
+                    )
+                ),
+                new CodeSample(
+'<?php 
+$a = 9000; 
+if (true) {
+    $foo = $bar;
+}',
+                    array(
+                        'if',
+                    )
+                ),
+            new CodeSample(
+                '<?php 
+                    
+if (true) {
+    $foo = $bar;
+    return;
+}',
+                array(
+                    'return',
+                )
+            ),
+                new CodeSample(
+'<?php 
+$a = 9000;                    
+switch ($a) {
+    case 42:
+        break;
+}',
+                    array(
+                        'switch',
+                    )
+                ),
+                new CodeSample(
+'<?php 
+if (null === $a) {
+    $foo->bar();
+    throw new \UnexpectedValueException("A cannot be null");
+}',
+                    array(
+                        'throw',
+                    )
+                ),
+                new CodeSample(
+'<?php 
+$a = 9000;                    
+try {
+    $foo->bar();
+} catch (\Exception $exception) {
+    $a = -1;
+}',
+                    array(
+                        'try',
+                    )
+                ),
+            ),
+            null,
+            sprintf(
+                'Provide an array with any of "%s".',
```

Comment:
I would rather make it an object instead of flat array.
like `['statements' => [try, continue]]` instead of just `[try, continue]`, so we will be able to add different options in the future (like eg "how many lines")